sdk/ai/azure-ai-agents/src/main/java/com/azure/ai/agents/implementation/http/HttpClientHelper.java

```diff
@@ -0,0 +1,230 @@
+
+    private static final ClientLogger LOGGER = new ClientLogger(HttpClientHelper.class);
+
+    private HttpClientHelper() {
+    }
+
+    /**
+     * Implements the OpenAI {@link HttpClient} interface that sends the HTTP request through the Azure HTTP pipeline.
+     * All requests and responses are converted on the fly.
+     *
+     * @param httpPipeline The Azure HTTP pipeline that will execute HTTP requests.
+     * @return A bridge client that honors the OpenAI interface but delegates execution to the Azure pipeline.
+     */
+    public static HttpClient mapToOpenAIHttpClient(HttpPipeline httpPipeline) {
+        return new HttpClientWrapper(httpPipeline);
+    }
+
+    private static final class HttpClientWrapper implements HttpClient {
+
+        private final HttpPipeline httpPipeline;
+
+        private HttpClientWrapper(HttpPipeline httpPipeline) {
+            this.httpPipeline = Objects.requireNonNull(httpPipeline, "'httpPipeline' cannot be null.");
+        }
+
+        @Override
+        public void close() {
+            // no-op
+        }
+
+        @Override
+        public HttpResponse execute(HttpRequest request) {
+            return execute(request, RequestOptions.none());
+        }
+
+        @Override
+        public HttpResponse execute(HttpRequest request, RequestOptions requestOptions) {
+            Objects.requireNonNull(request, "request");
+            Objects.requireNonNull(requestOptions, "requestOptions");
+
+            com.azure.core.http.HttpRequest azureRequest = buildAzureRequest(request);
+
+            Context requestContext = Context.NONE;
+            Timeout timeout = requestOptions.getTimeout();
+            if (timeout != null && !timeout.read().isZero() && !timeout.read().isNegative()) {
+                requestContext = requestContext.addData("azure-response-timeout", timeout.read());
+            }
+
+            com.azure.core.http.HttpResponse azureResponse = this.httpPipeline.sendSync(azureRequest, requestContext);
+            return new AzureHttpResponseAdapter(azureResponse);
+        }
+
+        @Override
+        public CompletableFuture<HttpResponse> executeAsync(HttpRequest request) {
+            return executeAsync(request, RequestOptions.none());
+        }
+
+        @Override
+        public CompletableFuture<HttpResponse> executeAsync(HttpRequest request, RequestOptions requestOptions) {
+            Objects.requireNonNull(request, "request");
+            Objects.requireNonNull(requestOptions, "requestOptions");
+
+            final com.azure.core.http.HttpRequest azureRequest = buildAzureRequest(request);
+
+            Context requestContext = new Context("azure-eagerly-read-response", true);
+            Timeout timeout = requestOptions.getTimeout();
+            if (timeout != null && !timeout.read().isZero() && !timeout.read().isNegative()) {
+                requestContext = requestContext.addData("azure-response-timeout", timeout.read());
+            }
```

Comment:
Add comments here to explain what this is doing and also mention the current limitation with a link to the feature request for `azure-core-http-<netty/okhttp>` to support connect, read and write timeouts per request.